### DISCUSSION TITLE
Watching Alpine store variables

### DISCUSSION THREAD

**Discussion by hrr-amboss** [alpinejs/alpine#2132] [4. General]
I'm trying to build an SPA with Alpine (yes 😬 ) and am registering app navigation in Alpine.store(). I'd like to scroll the window to the top to simulate a page reload. Therefore I'd like to use something like $watch but for Alpine store data, not for x-data elements. Is there any way to do this in Alpine?

**Comment by danddanddand** [comment]
https://ryangjchandler.co.uk/posts/how-to-watch-alpine-stores-for-changes

**Comment by hrr-amboss** [comment]
Awesome, thanks!

**Comment by Tim-Wils** [comment]
@hrr-amboss This can also be done with $watch('$store.storename.keyname') like so: https://jsfiddle.net/1v2L7wdq/

**Reply by hrr-amboss** [reply]
@Tim-Wils thanks so much. I had tried this but must have made an error and then thought it wouldn't be possible. Alpine is so awesome!
